docs/plugins.md

```diff
@@ -0,0 +1,215 @@
+# Authoring Plugins
+
+A published plugin must be its own installable `npm` module, following the
+plugin [naming convention]. You may also create [one-off plugins] for an app.
+
+A plugin's main export has the following shape:
+
+```js
+export default {
+  name: 'awesome',
+  dependencies: [/* list of plugin names */],
+  hooks: {
+    hookA(gasket) {
+
+    },
+    async hookB(gasket) {
+
+    },
+    hookC: {
+      timing: {},
+      handler(gasket) {
+
+      }
+    }
+  }
+};
+```
+
+The optional `dependencies` array lists other plugins that must be installed as
+a prerequisite. The `hooks` map has a key for every event the plugin handles,
+with the values being either the function itself or an object specifying
+`timing` options and the `handler`.
+
+## Naming convention
+
+Plugins and presets should adhere to the project-type prefixed naming convention. This formatting allows user plugins to be referenced with short names and will help avoid collisions. This convention also mimics those by other projects such a `@babel` and `@oclif`
+
+## Hooks
+
+Each `handler` function assigned to an event is invoked by the plugin engine
+when an event occurs. If an event is used for collection of data, each callback
+should return data. If the handling is asynchronous, the callback should return
+a `Promise`.
+
+If multiple plugins hook the same event, they'll run in parallel if async or in
+arbitrary order if synchronous. If a plugin wants to specify more specific
+sequencing, the hook should be an object with a timing property and handler
+function instead of a plain function. For example, this plugin contains an
+`init` hook that specifically runs *after* the `metadata` plugin:
+
+```js
+export default {
+  name: 'will-run-after-metadata',
+  hooks: {
+    init: {
+      timing: {
+        after: [ 'metadata' ]
+      },
+      handler: async function initHook(gasket) {
+        const { config } = gasket;
+        console.log(config.name);
+      }
+    }
+  }
+}
+```
```

Comment:
This is another outdated example. Also, we do not support "short plugin names" in the timing part. Can you spin up a different, more accurate example?